src/core/meshcop/tcat_agent.hpp

```diff
@@ -0,0 +1,398 @@
+        kStatusHashError = OT_TCAT_STATUS_HASH_ERROR, ///< The hash value presented by the commissioner was incorrect
+        kStatusUnauthorized =
+            OT_TCAT_STATUS_UNAUTHORIZED, ///< Sender does not have sufficient authorization for the given command
+    };
+
+    /**
+     * Represents TCAT message type.
+     *
+     */
+    enum TcatMessageType : uint8_t
+    {
+        kMessageTypeNone = OT_TCAT_MESSAGE_TYPE_NONE, ///< Message which has been sent without activating the TCAT agent
+        kMessageTypeStatus =
+            OT_TCAT_MESSAGE_TYPE_STATUS, ///< Message containing a status code (byte) as defined in otTcatStatusCode
+        kMessageTypeUdp = OT_TCAT_MESSAGE_TYPE_UDP, ///< Message directed to a UDP service
+        kMessageTypeTcp = OT_TCAT_MESSAGE_TYPE_TCP, ///< Message directed to a TCP service
+        kMessageTypeChangedUdpService =
+            OT_TCAT_MESSAGE_TYPE_CHANGED_TO_UDP_SERVICE, ///< Client has changed to a UDP service
+        kMessageTypeChangedTcpService =
+            OT_TCAT_MESSAGE_TYPE_CHANGED_TO_TCP_SERVICE, ///< Client has changed to a TCP service
+    };
+
+    /**
+     * Represents a TCAT certificate V3 extension attribute (OID 1.3.6.1.4.1.44970.x).
+     *
+     */
+    enum TcatCertificateAttribute
+    {
+        kCertificateDomainName         = 1,
+        kCertificateAuthorizationField = 3,
+        kCertificateNetworkName        = 4,
+        kCertificateExtendedPanId      = 5,
+    };
+
+    /**
+     * Represents TCAT status.
+     *
+     */
+    enum State : uint8_t
+    {
+        kStateDisabled,
+        kStateEnabled,
+        kStateConnected,
+    };
+
+    /**
+     * Initializes the Joiner object.
+     *
+     * @param[in]  aInstance     A reference to the OpenThread instance.
+     *
+     */
+    explicit TcatAgent(Instance &aInstance);
+
+    /**
+     * Enables the TCAT protocol.
+     *
+     * @retval kErrorNone           Successfully started the TCAT agent.
+     * @retval kErrorInvalidArgs    The aVendorInfo is invalid.
+     *
+     */
+    Error Start(const VendorInfo      &aVendorInfo,
+                AppDataReceiveCallback aAppDataReceiveCallback,
+                JoinCallback           aHandler,
+                void                  *aContext);
+
+    /**
+     * Stops the TCAT protocol.
+     *
+     */
+    void Stop(void);
+
+    /**
+     * Indicates whether or not the TCAT agent is enabled.
+     *
+     * @retval TRUE   The TCAT agent is enabled.
+     * @retval FALSE  The TCAT agent is not enabled.
+     *
+     */
+    bool IsEnabled(void) const { return mState != kStateDisabled; }
+
+    /**
+     * Indicates whether or not the TCAT agent is connected.
+     *
+     * @retval TRUE   The TCAT agent is connected.
+     * @retval FALSE  The TCAT agent is not connected.
+     *
+     */
+    bool IsConnected(void) const { return mState == kStateConnected; }
+
+    /**
+     * Indicates whether or not a command class with given flags is authorized.
```

Comment:
Comments should be updated to clarify what the `aDataset` parameter provides, or how the command class is extracted from `aDataset`? This isn't obvious. And there is no `aCommandClass` parameter to be checked for authorization.

Reply:
(to discuss this item later on)  It seems that a few functions are missing the @param tags in the comment header. That might be an OT requirement for the definition of the API for class TcatAgent . Checking @abtink do you know? It would certainly be helpful here to have all the @param tags present.

Reply:
@canisLupus1313 Add param description.

Reply:
@canisLupus1313 add params description